benchmark_runner/common/elasticsearch/elasticsearch_operations.py

```diff
@@ -201,10 +198,99 @@ def update_elasticsearch_index(self, index: str, id: str, metadata: dict = ''):
     @logger_time_stamp
     def get_elasticsearch_index_by_id(self, index: str, id: str):
         """
-        This method return elastic search index data by id
+        This method return elasticsearch index data by id
         :param index: index name
         :param id: The specific index id
         :return:
         """
         return self.__es.get(index=index, id=id)
 
+    @typechecked()
+    @logger_time_stamp
+    def delete_elasticsearch_index_by_id(self, index: str, id: str):
+        """
+        This method delete elasticsearch index data by id
+        :param index: index name
+        :param id: The specific index id
+        :return:
+        """
+        return self.__es.delete(index=index, id=id)
+
+    @typechecked()
+    @logger_time_stamp
+    def get_query_data_between_dates(self, start_datetime: datetime, end_datetime: datetime):
+        """
+        This method returns the query for fetching data between dates
+        @param start_datetime:
+        @param end_datetime:
+        @return:
+        """
+        query = {
+                "bool": {
+                    "filter": {
+                        "range": {
+                            "timestamp": {
+                                "format": "yyyy-MM-dd HH:mm:ss"
+                            }
+                        }
+                    }
+                }
+        }
+        query['bool']['filter']['range']['timestamp']['lte'] = str(end_datetime.replace(microsecond=0))
+        query['bool']['filter']['range']['timestamp']['gte'] = str(start_datetime.replace(microsecond=0))
+        return query
+
+    @typechecked()
+    @logger_time_stamp
+    def get_index_ids_between_dates(self, index: str, start_datetime: datetime = None, end_datetime: datetime = None):
+        """
+        This method return list of index ids between dates
+        @param index:
+        @param start_datetime:
+        @param end_datetime:
+        @return: list of ids
+        """
+        es_data = []
+        query = self.get_query_data_between_dates(start_datetime, end_datetime)
+        response = self.__es.search(index=index, query=query, size=100, scroll='1h')
```

Comment:
Does this limit the number of results to 100?

Reply:
It will fetch 100 rows in each page

Reply:
OK.